spec/controllers/admin/sessions_controller_spec.rb

```diff
@@ -0,0 +1,69 @@
+require "rails_helper"
+require_relative "shared_helpers"
+
+RSpec.describe Admin::SessionsController, type: :controller do
+  let(:admin){create(:user, role: :admin)}
+
+  def password_login user, options = {}
+    post :create, params: {login: {email: options.fetch(:email, user.email),
+                                   password: options.fetch(:password, user.password)}}
+  end
+
+  describe "routing" do
+    it{should route(:get, "/admin/login").to(action: :new)}
+    it{should route(:post, "/admin/login").to(action: :create)}
+    it{should route(:get, "/admin/logout").to(action: :destroy)}
+  end
+
+  describe "GET #new" do
+    it "renders the login view if user not logged in" do
+      get :new
+      expect(response).to render_template("new")
+    end
```

Comment:
Chỗ này em thêm thay đổi thành 
```suggestion
context "when user not logged in"
    it "should renders the login view " do
      get :new
      expect(response).to render_template("new")
    end
    end 
```

Reply:
THay đổi lại với những trường hợp tương tự nhé